### ISSUE TITLE
ttnn.nonzero indices output buffer is 4x larger than declared shape; values only at stride-4 offset 3

### ISSUE THREAD

**Issue by Aswinmcw** [tenstorrent/tt-metal#48939]
## Summary

`ttnn::prim::nonzero` returns `(count, indices)`. The `indices` tensor is declared in `compute_output_specs` with `input.logical_shape()` and `DataType::UINT32`, so for a `(1,1,1,32)` `float32` input we expect 32 uint32 slots (128 bytes). What we actually get back is a buffer of **128 uint32 slots (512 bytes)** — 4x larger than declared — with the real non-zero indices only appearing at positions `buf[3::4]` (stride 4, offset 3). All other positions contain uninitialized / stale data.

The `count` output (hardcoded to shape `(1,1,1,8)`) is unaffected.

## Repro

```python
import torch
import ttnn

device = ttnn.open_device(device_id=0)

# 32-element float32 input with non-zeros at positions 5, 10, 20
x = torch.zeros(1, 1, 1, 32, dtype=torch.float32)
x[0, 0, 0, 5]  = 1.0
x[0, 0, 0, 10] = 2.0
x[0, 0, 0, 20] = 3.0

t = ttnn.from_torch(x, dtype=ttnn.float32, layout=ttnn.ROW_MAJOR_LAYOUT, device=device)
count_t, idx_t = ttnn.nonzero(t)

count = ttnn.to_torch(count_t).flatten()[0].item()   # 3
idx   = ttnn.to_torch(idx_t).flatten()

print(f"count       = {count}")
print(f"idx.numel() = {idx.numel()}")   # 128 -- expected 32
print(f"idx[:count] = {idx[:count].tolist()}")           # [garbage, garbage, garbage]
print(f"idx[3::4][:count] = {idx[3::4][:count].tolist()}") # [5, 10, 20]  <-- actual data lives here
```

### Expected

```
count       = 3
idx.numel() = 32
idx[:count] = [5, 10, 20]
```

### Actual

```
count       = 3
idx.numel() = 128
idx[:count] = [<uninitialized>, <uninitialized>, <uninitialized>]
idx[3::4][:count] = [5, 10, 20]
```

Same 4x factor observed with other last-dim sizes.

## Root cause hypothesis

Two suspect spots — I have not confirmed which one owns the bug, but both handle the input-vs-output element size incorrectly:

1. **`compute_output_specs`** in `ttnn/cpp/ttnn/operations/data_movement/non_zero_indices/device/non_zero_indices_device_operation.cpp:29-33`:
   ```cpp
   TensorLayout layout(DataType::UINT32, PageConfig(Layout::ROW_MAJOR), args.output_memory_config);
   return {TensorSpec(num_non_zero_shape, layout),
           TensorSpec(tensor_args.input.logical_shape(), layout)};  // <-- indices spec
   ```
   Declared as UINT32 with `input.logical_shape()`, which by declaration is 32 uint32 = 128 bytes. But the underlying allocation appears to inherit the input's per-element byte count somewhere downstream, yielding 4x the storage for a float32 input.

2. **`non_zero_indices_program_factory.cpp:32` / kernel `non_zero_indices_sc_reader.cpp:32`**:
   ```cpp
   const auto out1 = TensorAccessor(dst1_args, output_addr_1, aligned_elements * element_size);
   ```
   `element_size` is the runtime arg set to `input.element_size()` (line 76 of the program factory), i.e. 4 for float32. The output tensor is UINT32 which happens to also be 4 bytes/elem, so byte-count works out for float32 — but for smaller input dtypes (e.g. bfloat16, uint8) this will alias the wrong page size against the uint32 indices buffer.

The stride-4-offset-3 pattern (values at `buf[3::4]` in a 128-uint32 buffer) is consistent with the kernel doing 32 uint32 writes into a NoC page of 32*4 uint32 slots where each write lands at the last uint32 of a 4-uint32 page. Something in the TensorAccessor/page-size setup is scaling by `element_size` (input) when the destination is uint32.

## Impact

Any user of `ttnn.nonzero` on `float32`/`bfloat16` inputs has to know the magic `[3::4]` (or dtype-dependent equivalent) offset to extract indices. Torch parity wrappers (e.g. `torch.where(x)` single-arg form) built on top of this cannot rely on the declared shape.

## Suggested fix direction

- Decouple output page-size / accessor from `input.element_size()`; use the actual output dtype's element size (4 for UINT32) when computing `aligned_elements * elem_bytes` for the destination TensorAccessor.
- Verify `compute_output_specs` produces a buffer sized exactly `logical_shape().volume() * sizeof(uint32_t)` and not scaled by the input dtype.
- Add a small unit test that checks `idx.numel() == input.numel()` and `idx[:count] == expected_indices` for at least `float32`, `bfloat16`, and `uint8` inputs.

## Environment

- Wormhole N150 dev box, `ttnn` at repo HEAD.

**Comment by Aswinmcw** [comment]
Closing — my diagnosis was wrong. The `[1,1,1,N*4]` layout with values at stride-4 is intentional: each non-zero produces a `(b, n, h, c)` 4-tuple of uint32 values, matching `torch.nonzero(x, as_tuple=False)`. Documented in the existing test at `tests/ttnn/unit_tests/operations/data_movement/test_non_zero_indices.py:19-21` and the kernel comments. Not a bug. Sorry for the noise.

**Comment by Aswinmcw** [comment]
Testing comment via BrAInClaw
